Core/src/Geometry/GeometryIdentifierBlueprintNode.cpp

```diff
@@ -0,0 +1,238 @@
+        throw std::logic_error("Volume already has a volume ID");
+      }
+      ACTS_DEBUG(prefix << "~> Setting volume ID to " << m_volumeId
+                        << " for volume " << v.volumeName());
+      v.assignGeometryId(v.geometryId().withVolume(m_volumeId));
+    });
+  }
+
+  const std::string& name() const override { return m_name; }
+
+ private:
+  GeometryIdentifier::Value m_volumeId;
+  std::string m_name;
+};
+
+}  // namespace
+
+struct GeometryIdentifierBlueprintNodeImpl {
+  void add(std::unique_ptr<Configuration> configuration) {
+    m_configurations.push_back(std::move(configuration));
+
+    std::vector<std::string> names;
+    for (auto& conf : m_configurations) {
+      names.push_back(conf->name());
+    }
+
+    m_name = boost::algorithm::join(names, ", ");
+  }
+
+  std::vector<std::unique_ptr<Configuration>> m_configurations;
+  std::string m_name;
+
+  GeometryIdentifierBlueprintNode::CompareVolumes m_sortBy;
+};
+
+GeometryIdentifierBlueprintNode::GeometryIdentifierBlueprintNode()
+    : m_impl(std::make_unique<GeometryIdentifierBlueprintNodeImpl>()) {}
+
+GeometryIdentifierBlueprintNode::~GeometryIdentifierBlueprintNode() = default;
+
+Volume& GeometryIdentifierBlueprintNode::build(const BlueprintOptions& options,
+                                               const GeometryContext& gctx,
+                                               const Logger& logger) {
+  if (children().size() != 1) {
+    throw std::invalid_argument(
+        "GeometryIdentifierBlueprintNode must have exactly one child");
+  }
+
+  if (m_impl->m_configurations.empty()) {
+    throw std::invalid_argument(
+        "GeometryIdentifierBlueprintNode has no configuration");
+  }
+
+  return children().at(0).build(options, gctx, logger);
+}
+
+PortalShellBase& GeometryIdentifierBlueprintNode::connect(
+    const BlueprintOptions& options, const GeometryContext& gctx,
+    const Logger& logger) {
+  return children().at(0).connect(options, gctx, logger);
+}
+
+void GeometryIdentifierBlueprintNode::finalize(const BlueprintOptions& options,
+                                               const GeometryContext& gctx,
+                                               TrackingVolume& parent,
+                                               const Logger& logger) {
+  ACTS_DEBUG(prefix() << "Finalizing geo id " << name() << " with parent "
+                      << parent.volumeName());
+  std::set<const TrackingVolume*> previous;
+  std::ranges::for_each(parent.volumes(),
+                        [&](const auto& v) { previous.insert(&v); });
+
+  children().at(0).finalize(options, gctx, parent, logger);
+
+  std::vector<TrackingVolume*> volumes;
+  for (auto& v : parent.volumes()) {
+    // Skip volumes that were already in the parent before the subtree
+    // was processed
+    if (previous.contains(&v)) {
+      continue;
+    }
+
+    volumes.push_back(&v);
+  }
+
+  if (m_impl->m_sortBy) {
```

Comment:
How is this implicitly converted into `bool`?

Reply:
`std::function` declares an [`operator bool`](https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool), and I believe it's actually the only way to check if it contains a value.